webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/inventory/InventoryFileWatcher.scala

```diff
@@ -391,17 +418,57 @@ class ProcessFile(
   }
 
   //start the process
-  ZioRuntime.internal.unsafeRunSync(loop.forkDaemon)
+  ZioRuntime.internal.unsafeRunSync(processMessage().forever.forkDaemon)
 
   def addFilePure(file: File): UIO[Unit] = {
-    queue.offer(WatchEvent.Mod(file)).unit
+    watchEventQueue.offer(WatchEvent.Mod(file)).unit
   }
 
   def addFile(file: File): Unit = {
     ZioRuntime.internal.unsafeRunSync(addFilePure(file))
   }
 
 
+  /*
+   * This is a blocking action, only call it with a non-blocking buffer in front of it.
+   */
+
+  protected def sendToProcessorBlocking(inventory: File, signature: Option[File]): UIO[Unit] = {
+    // we don't manage race condition very well, so we have cases where
+    // we can have two things trying to move
+    def safeMove[T](chunk: =>T): IOResult[Unit] = {
+      Task.effect{chunk ; ()}.catchAll {
+        case ex: NoSuchFileException => // ignore
+          InventoryProcessingLogger.debug(s"Ignored exception '${ex.getClass.getSimpleName} ${ex.getMessage}'. The was file was correctly handled.")
```

Comment:
The error message is not very explicit
"The file '${inventory.pathAsString}' was correctly handled" maybe?

Reply:
Arf :)